src/virtualship/expedition/waypoint.py

```diff
@@ -1,16 +1,28 @@
 """Waypoint class."""
 
-from dataclasses import dataclass
 from datetime import datetime
 
+from pydantic import BaseModel, field_serializer
+
 from ..location import Location
 from .instrument_type import InstrumentType
 
 
-@dataclass
-class Waypoint:
+class Waypoint(BaseModel):
     """A Waypoint to sail to with an optional time and an optional instrument."""
 
     location: Location
     time: datetime | None = None
     instrument: InstrumentType | list[InstrumentType] | None = None
+
+    @field_serializer("instrument")
+    def serialize_instrument(self, instrument):
+        """Ensure InstrumentType is serialized as a string (or list of strings)."""
+        if isinstance(instrument, list):
+            return [inst.value for inst in instrument]
+        return instrument.value if instrument else None
+
+    @field_serializer("time")
+    def serialize_time(self, time):
+        """Ensure datetime is formatted properly in YAML."""
+        return time.strftime("%Y-%m-%d %H:%M:%S") if time else None
```

Comment:
I'm not 100% sure what this code change is. Is there a need to have specific field serializers for these? I imagine that the default Pydantic model serialization would be sufficient?

Also, the changing from `dataclass` from `BaseModel` changes the init signature requiring things to be passed in via kwargs and not args (which isn't a problem, but just wanting to get a better picture of things before shipping).

Reverting for now, but we can put in a furture PR :)

Reply:
i see why this was done now - that InstrumentType serialization wasn't happening as expected. Ok, let's keep this (though still not sure about the time serialization as that looks to be ok?)